hudi-client/hudi-client-common/src/test/java/org/apache/hudi/utils/HoodieWriterClientTestHarness.java

```diff
@@ -554,7 +554,7 @@ protected List<HoodieRecord<RawTripTestPayload>> dedupForCopyOnWriteStorage(Hood
         .getReaderContextFactoryForWrite(metaClient, HoodieRecord.HoodieRecordType.AVRO, writeConfig.getProps()).getContext();
     List<String> orderingFieldNames = getOrderingFieldNames(
         readerContext.getMergeMode(), writeClient.getConfig().getProps(), metaClient);
-    RecordMergeMode recordMergeMode = HoodieTableConfig.inferCorrectMergingBehavior(null, writeConfig.getPayloadClass(), null,
+    RecordMergeMode recordMergeMode = HoodieTableConfig.inferMergingConfigs(metaClient.getTableConfig().getRecordMergeMode(), writeConfig.getPayloadClass(), null,
```

Comment:
Should `metaClient.getTableConfig().getRecordMergeMode()` directly be used instead of calling `inferMergingConfigs`?  Trying to understand the intention here in test.